crates/net/network/src/transactions.rs

```diff
@@ -431,10 +478,19 @@ where
         match cmd {
             TransactionsCommand::PropagateHash(hash) => self.on_new_transactions(vec![hash]),
             TransactionsCommand::PropagateHashTo(_hash, _peer) => todo!(),
-            TransactionsCommand::GetActivePeers => todo!(),
-            TransactionsCommand::PropagateTransactionsTo(_txs, _peer) => todo!(),
-            TransactionsCommand::GetTransactionHashes(_peers) => todo!(),
-            TransactionsCommand::GetPeerTransactionHashes(_peer) => todo!(),
+            TransactionsCommand::GetActivePeers => {
+                self.get_active_peers();
+            }
+            TransactionsCommand::PropagateTransactionsTo(_txs, _peer) => {
+                let propagated = self.propagate_transactions_to_peer(_txs, _peer);
+                self.pool.on_propagated(propagated);
+            }
+            TransactionsCommand::GetTransactionHashes(_peers) => {
+                self.get_transaction_hashes(_peers);
```

Comment:
this can be:

`self.peers.get(&peer).map(|p|p.transactions.iter().copied().collect::<HashSet<_>>()).unwrap_or_default()`